corehq/apps/change_feed/consumer/feed.py

```diff
@@ -100,7 +100,7 @@ def get_current_checkpoint_offsets(self):
         latest_offsets = self.get_latest_offsets()
         ret = {}
         for topic_partition, sequence in self.get_processed_offsets().items():
-            if sequence == latest_offsets[topic_partition]:
+            if sequence >= latest_offsets[topic_partition]:
```

Comment:
This was more of just a logical correction covering a case that previously it wasn't thinking about, but which probably never happens in production; previously if through any combination of wiping kafka and changing checkpoints the checkpoint was ever greater than the latest offset, it would just happily keep incrementing it by one each time nonsensically. Now it moves to the latest offset. Maybe the right thing would be to have it raise an error? I can run these changes locally without this change and it doesn't seem to cause any issues. Happy to revert just this portion or revisit if you have suggestions, but I'm not sure it's worth adding a test for.

Reply:
My default reaction to "probably never happens in production" is not usually "not sure it's worth adding a test for." I would either determine that it _definitely never happens in production_ or write a test. I'll leave it to your judgement.

Reply:
Just had a call with Cal where we traced an ICDS "pillow rewind" issue potentially back to this. (The hypothesis is that this + kafka temporarily returning 0 for partitions that are offline caused the rewind.)

I'm gonna score that as Daniel 1, Danny 0, and it's a good reminder not to throw extraneous things I don't fully understand and can't test into a larger PR.

Reply:
https://github.com/dimagi/commcare-hq/pull/26899